specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-07-01/virtualMachineImage.json

```
@@ -789,6 +789,90 @@
       }
     }
   },
+  "x-ms-paths":{
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/publishers/{publisherName}/artifacttypes/vmimage/offers/{offer}/skus/{skus}/versions?$expand=Properties": 
+    {
+      "get":{ 
+        "operationId": "VirtualMachineImages_ListWithProperties",
+        "parameters": [
+          {
+            "name": "location",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of a supported Azure region."
+          },
+          {
+            "name": "publisherName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "A valid image publisher."
+          },
+          {
+            "name": "offer",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "A valid image publisher offer."
+          },
+          {
+            "name": "skus",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "A valid image SKU."
+          },
+          {
+            "name": "$expand",
+            "in": "query",
+            "required": false,
```

```suggestion
            "required": true,
```

Given the scope of this operation is specifically to be sure this query parameter is provided, I would make it true, to be sure autorest process it as an internal constant, and do not expose it as a parameter that may take `null` as a valid value. It may work well like that, but better safe than sorry :)

---

That makes senses thanks for pointing out.